doc/service-integrations.md

```diff
@@ -0,0 +1,70 @@
+# Integration with upstream services
+
+## Overview
+
+In a nutshell, the Shields Badge Server handles the responsibilities of accepting requests for badges, and then serving those badges back to users.
+
+A grossly oversimplified visualization would probably look something like this:
+
+```mermaid
+sequenceDiagram
+    actor User
+    participant B as Badge Server
+    participant P as Data Provider
+    User->>+B: I'd like a badge for my project please!
+    B->>+P: Get data for project
+    P-->>-B: Data
+    B->>B: Make badge with data
+    B-->>-User: Here's your badge!
+```
```

Comment:
I've been looking for an excuse to play with this ever since GitHub [announced embedded support](https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/), rendered version can be found [here](https://github.com/badges/shields/blob/d24e055ec821cef365e279b1ca2b1c82afb48aec/doc/service-integrations.md#overview).

As is often the case with diagrams as code, they aren't quite as pretty and pixel perfect as human drawn ones would be, but i think something simple and embedded here is worthwhile to have a visual

Reply:
:art: